src/index.tsx

```diff
@@ -172,16 +172,31 @@ function useWavesurferEvents(wavesurfer: WaveSurfer | null, events: OnWavesurfer
   }, [wavesurfer, ...flatEvents])
 }
 
+function useWavesurferLoad(
+  wavesurfer: WaveSurfer | null,
+  url: string | undefined,
+  peaks: (Float32Array | number[])[] | undefined,
+  duration: number | undefined,
+) {
+  useEffect(() => {
+    if (!wavesurfer) return
+    if (!url) return
+
+    wavesurfer.load(url, peaks, duration)
+  }, [wavesurfer, url, peaks, duration])
+}
+
 /**
  * Wavesurfer player component
  * @see https://wavesurfer.xyz/docs/modules/wavesurfer
  * @public
  */
 const WavesurferPlayer = memo((props: WavesurferProps): ReactElement => {
   const containerRef = useRef<HTMLDivElement | null>(null)
-  const [options, events] = useWavesurferProps(props)
+  const [{ url, peaks, duration, ...options }, events] = useWavesurferProps(props)
```

Comment:
`...options` will probably create a new object on every render and should be memoized with useMemo.